test/unit/services/dataAccessLayer.test.js

```diff
@@ -0,0 +1,62 @@
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+
+const userQuery = require("../../../models/users");
+const sinon = require("sinon");
+const { retrieveUsers, removeSensitiveInfo } = require("../../../services/dataAccessLayer");
+const userData = require("../../fixtures/user/user")();
+const { sensitiveData } = require("../../../constants/users");
+chai.use(chaiHttp);
+const expect = chai.expect;
+
+describe("Data Access Layer", function () {
+  describe("retrieveUsers", function () {
+    it("should fetch a single user by ID and remove sensitive info", async function () {
+      const fetchUserStub = sinon.stub(userQuery, "fetchUser");
+      fetchUserStub.returns(Promise.resolve({ user: userData[12] }));
```

Comment:
what will happen if multiple id's are passed

Reply:
multiple ids would not be passed in this case as per the function signature of retrieveUsers